libc/string_lenchr.c

```diff
@@ -150,9 +169,259 @@ TEST(string_len, big)
+	TEST_ASSERT_EQUAL_STRING("", strrchr(torn, '\0'));
+	TEST_ASSERT_EQUAL_STRING("", memchr(torn, '\0', sz));
+
+	TEST_ASSERT_EQUAL_STRING("foo", strchr(torn, 'f'));
+	TEST_ASSERT_EQUAL_STRING("foo", strrchr(torn, 'f'));
+	TEST_ASSERT_EQUAL_STRING("foo", memchr(torn, 'f', sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(torn, 'b'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(torn, 'b'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(torn, 'b', sz));
+
+	/* 99%# decode ASCII after number 9 which is not in ASCII chain in this case */
+	TEST_ASSERT_EQUAL_STRING("99%#", strchr(format, '9'));
+	TEST_ASSERT_EQUAL_STRING("9%#", strrchr(format, '9'));
+	TEST_ASSERT_EQUAL_STRING("99%#", memchr(format, '9', 7));
+
+	/* Testing if we can get a number from ASCII if it represents the symbol of him */
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(format, '4'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(format, '4'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(format, '4', 7));
+
+	/* Checking if we can get a white sign from a string full of different white signs */
+	sz = ptrContentSize(whites);
+	TEST_ASSERT_EQUAL_STRING("\n", strchr(whites, '\n'));
+	TEST_ASSERT_EQUAL_STRING("\n", strrchr(whites, '\n'));
+	TEST_ASSERT_EQUAL_STRING("\n", memchr(whites, '\n', sz));
+
+	/* Checking possibility and correction of value after white signs */
+	sz = ptrContentSize(xwhites);
+	TEST_ASSERT_EQUAL_STRING("x", strchr(xwhites, 'x'));
+	TEST_ASSERT_EQUAL_STRING("x", strrchr(xwhites, 'x'));
+	TEST_ASSERT_EQUAL_STRING("x", memchr(xwhites, 'x', sz));
+
+	TEST_ASSERT_EQUAL_STRING(whites, strchr(whites, ' '));
+	TEST_ASSERT_EQUAL_STRING(whites, strrchr(whites, ' '));
+	TEST_ASSERT_EQUAL_STRING(whites, memchr(whites, ' ', strlen(whites)));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(whites, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(whites, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(whites, 'x', strlen(whites)));
+}
+
+
+TEST(string_chr, not_ascii)
+{
+	char *notAsciiString = "♥♣♠◊⊗こんにちは❉❉⌨⌨⌨⌨⌨⌨⌨⌨❉❉♦x";
+	char notAsciiSet[129];
+	int sz, i;
+
+	/* Checking ability to read the chars out of ASCII charset */
+	sz = ptrContentSize(notAsciiString);
+	TEST_ASSERT_EQUAL_STRING("x", strchr(notAsciiString, 'x'));
+	TEST_ASSERT_EQUAL_STRING("x", strrchr(notAsciiString, 'x'));
+	TEST_ASSERT_EQUAL_STRING("x", memchr(notAsciiString, 'x', sz));
+
+	/* Checking  out of ASCII bytes */
+	for (i = 128; i <= 255; i++) {
+		notAsciiSet[i - 128] = i;
+	}
+	notAsciiSet[128] = 0;
+
+	/* Testing capability of functions to hold and read not ASCII set */
+	sz = sizeof(notAsciiSet) - 1;
+	TEST_ASSERT_EQUAL_STRING(notAsciiSet, strchr(notAsciiSet, notAsciiSet[0]));
+	TEST_ASSERT_EQUAL_STRING(notAsciiSet, strrchr(notAsciiSet, notAsciiSet[0]));
+	TEST_ASSERT_EQUAL_STRING(notAsciiSet, memchr(notAsciiSet, notAsciiSet[0], sz));
+}
+
+
+TEST(string_chr, empty)
+{
+	char *empty = "";
+	int sz;
+
+	/* Checking if we can get an empty string */
+	sz = 1;
+	TEST_ASSERT_EQUAL_STRING("", strchr(empty, '\0'));
```

Comment:
In my opinion it's not needed to declare empty variable, we can just pass "" everywhere.

```suggestion
	TEST_ASSERT_EQUAL_STRING("", strchr("", '\0'));
```